drivers/staging/vc04_services/bcm2835-isp/bcm2835-v4l2-isp.c

```diff
@@ -1720,25 +1732,51 @@ static int bcm2835_isp_probe(struct platform_device *pdev)
 			  BCM2835_ISP_NUM_OUTPUTS,
 			  dev->component->outputs,
 			  BCM2835_ISP_NUM_CAPTURES + BCM2835_ISP_NUM_METADATA);
-		goto error;
+		return -EINVAL;
 	}
 
 	atomic_set(&dev->num_streaming, 0);
 
 	for (i = 0; i < BCM2835_ISP_NUM_NODES; i++) {
 		struct bcm2835_isp_node *node = &dev->node[i];
 
-		ret = register_node(dev, node, i);
+		ret = register_node(dev, instance, node, i);
 		if (ret)
-			goto error;
+			return ret;
 	}
 
-	ret = media_controller_register(dev);
+	ret = media_controller_register(dev, instance);
 	if (ret)
-		goto error;
+		return ret;
+
+	return 0;
+}
+
+static int bcm2835_isp_remove(struct platform_device *pdev)
+{
+	unsigned int i;
+
+	for (i = 0; i < ARRAY_SIZE(bcm2835_isp_instance); i++) {
+		if (bcm2835_isp_instance[i])
+			bcm2835_isp_remove_instance(bcm2835_isp_instance[i]);
+	}
+
+	return 0;
+}
+
+static int bcm2835_isp_probe(struct platform_device *pdev)
+{
+	unsigned int i;
+	int ret;
+
+	for (i = 0; i < ARRAY_SIZE(bcm2835_isp_instance); i++) {
+		ret = bcm2835_isp_probe_instance(pdev, &bcm2835_isp_instance[i],
+						 i);
+		if (ret)
+			goto error;
```

Comment:
If it fails with i > 0, then we need to iteratively call bcm2835_isp_remove_instance on those that were created successfully. If we haven't updated ```bcm2835_isp_instance[i]```, then calling ```bcm2835_isp_remove``` will do that for us.